R/data-ddl-utils.R

```diff
@@ -0,0 +1,99 @@
+#' Function runs the `code`, masks the `code` and creates `teal_data` object.
+#' @param data (`teal_data`) object
+#' @param code (`language`) code to evaluate
+#' @param input (`list`) containing inputs to be used in the `code`
+#' @param input_mask (`list`) containing inputs to be masked in the `code`
+#'
+#' @return `teal_data` object
+#'
+#' @export
+eval_and_mask <- function(data,
+                          code,
+                          input = list(),
+                          input_mask = list()) {
+  # todo: do we need also within_and_mask?
+  checkmate::assert_list(input)
+  if (inherits(input, "reactivevalues")) {
+    input <- shiny::reactiveValuesToList(input)
+  }
```

Comment:
you can do `shiny::reactiveValuesToList` on function call and expect a list - will be simpler